pynamodb/models.py

```diff
@@ -460,6 +490,11 @@ def save(self, condition=None, conditional_operator=None, **expected_values):
             kwargs.update(expected=self._build_expected_values(expected_values, PUT_FILTER_OPERATOR_MAP))
         kwargs.update(conditional_operator=conditional_operator)
         kwargs.update(condition=condition)
+
+        if in_transaction is not None:
```

Comment:
An issue we're running into with this kind of implementation-

What will happen if we try to overload the `save` method with additional validation logic?

1st save: Validate on non-existence of a field, then we add the field for saving. Because it's a transaction, we don't actually commit.
2nd save via transaction: Validate on non-existence of a field, but this field now exists. Validation fails.

Reply:
^ can ignore this now; as @hallie pointed out, it's actually caused by a manual retry. Can use the retries on the transaction to implement retry logic.